tests/vmi_ignition_test.go

```diff
@@ -0,0 +1,107 @@
+	cirrosUser     = "cirros"
+	cirrosPassword = "gocubsgo"
+	// rootCmd        = "%23%21/bin/sh%0Aecho%20password%20%7C%20passwd%20--stdin%20root%0A"
+	// motd           = "Welcome%20to%20Kubevirt%0A"
+)
+
+var _ = Describe("[rfe_id:151][crit:high][vendor:cnv-qe@redhat.com][level:component]IgnitionData", func() {
+
+	flag.Parse()
+
+	virtClient, err := kubecli.GetKubevirtClient()
+	tests.PanicOnError(err)
+
+	LaunchVMI := func(vmi *v1.VirtualMachineInstance) {
+		By("Starting a VirtualMachineInstance")
+		obj, err := virtClient.RestClient().Post().Resource("virtualmachineinstances").Namespace(tests.NamespaceTestDefault).Body(vmi).Do().Get()
+		Expect(err).To(BeNil())
+
+		By("Waiting the VirtualMachineInstance start")
+		_, ok := obj.(*v1.VirtualMachineInstance)
+		Expect(ok).To(BeTrue(), "Object is not of type *v1.VirtualMachineInstance")
+		Expect(tests.WaitForSuccessfulVMIStart(obj)).ToNot(BeEmpty())
+	}
+
+	VerifyIgnitionDataVMI := func(vmi *v1.VirtualMachineInstance, commands []expect.Batcher, timeout time.Duration) {
+		By("Expecting the VirtualMachineInstance console")
+		expecter, _, err := tests.NewConsoleExpecter(virtClient, vmi, 10*time.Second)
+		Expect(err).ToNot(HaveOccurred())
+		defer expecter.Close()
+
+		By("Checking that the VirtualMachineInstance serial console output equals to expected one")
+		resp, err := expecter.ExpectBatch(commands, timeout)
+		log.DefaultLogger().Object(vmi).Infof("%v", resp)
+		Expect(err).ToNot(HaveOccurred())
+	}
+
+	BeforeEach(func() {
+		tests.BeforeTestCleanup()
+	})
+
+	Describe("[rfe_id:151][crit:medium][vendor:cnv-qe@redhat.com][level:component]A new VirtualMachineInstance", func() {
+		Context("with IgnitionData annotation", func() {
+			Context("with injected ssh-key", func() {
+				It("[test_id:1616]should have ssh-key under authorized keys", func() {
+					os.Setenv("FEATURE_GATES", "ExperimentalIgnitionSupport")
+					vmi := tests.NewRandomVMIWithEphemeralDisk(tests.ContainerDiskFor(tests.ContainerDiskCirros))
+					// ignitionData := fmt.Sprintf("{ \"ignition\": { \"config\": {}, \"version\": \"2.2.0\" }, \"networkd\": {}, \"passwd\": { \"users\": [ { \"name\": \"core\", \"sshAuthorizedKeys\": [ \"%s\" ] } ] }, \"storage\": { \"files\": [ { \"contents\": { \"source\": \"data:%s\", \"verification\": {} }, \"filesystem\": \"root\", \"mode\": 420, \"path\": \"/etc/motd\" }, { \"contents\": { \"source\": \"data:%s\", \"verification\": {} }, \"filesystem\": \"root\", \"mode\": 448, \"path\": \"/root/first.sh\" } ] }, \"systemd\": { \"units\": [ { \"contents\": \"[Service]\nType=oneshot\nExecStart=/root/first.sh\n[Install]\nWantedBy=multi-user.target\n\", \"enabled\": true, \"name\": \"first-boot.service\" } ] } }", pubKey, motd, rootCmd)
+					ignitionData := "ignition injected"
+					vmi.Annotations = map[string]string{"kubevirt.io/ignitiondata": ignitionData}
+
+					LaunchVMI(vmi)
+
+					VerifyIgnitionDataVMI(vmi, []expect.Batcher{
```

Comment:
We have a function in the utils package that handles logging into cirros for us. look for LoggedInCirrosExpecter()

We prefer using a helper function for that because it means we just have to update the logic in one place if our cirros image changes the user/password or anything else during the startup process (which has actually happened)